pkg/commands/kustfile/kustomizationfile.go

```diff
@@ -176,10 +178,31 @@ func (mf *kustomizationFile) Write(kustomization *types.Kustomization) error {
 	if kustomization == nil {
 		return errors.New("util: kustomization file arg is nil")
 	}
+
 	data, err := mf.marshal(kustomization)
 	if err != nil {
 		return err
 	}
+
+	if filepath.Ext(mf.path) == ".json" {
+
+		// Unmarshal the YAML instead of directly marshaling to JSON to support future changes of kustomizationFile::marshal method:
+		k := make(map[string]interface{})
+		err := yaml.Unmarshal(data, &k)
+		if err != nil {
+			return err
+		}
+
+		// Remove `kind` and `apiVersion` so as make minimal changes to original source file:
+		delete(k, "kind")
+		delete(k, "apiVersion")
```

Comment:
When the original file contains `kind` and `apiVersion`, this will also remove them. 
Let's not delete those two fields.